compiler/onecc-docker/onecc-docker

```diff
@@ -0,0 +1,74 @@
+#!/usr/bin/env bash
+''''exec "python3" "$0" "$@" #'''
+
+# Copyright (c) 2022 Samsung Electronics Co., Ltd. All Rights Reserved
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import sys
+import subprocess
+import json
+import os
+import requests
+
+
+def _is_image(version) : 
+    cmd = ['docker', 'images', f'onecc:{version}']
+    p = subprocess.Popen(cmd,  stdout=subprocess.PIPE, stderr=subprocess.STDOUT) 
+    out, err = p.communicate()
+    dic = {}
+
+    decoded_out = [i.split() for i in out.decode().split("\n") if len(i)]
+
+    if len(decoded_out) <2 :
+        return False
+
+    dic[decoded_out[0][0]] = decoded_out[1][0]
+    dic[decoded_out[0][1]] = decoded_out[1][1]
+
+    tag_name = cmd[2].split(':')
+
+    if dic['REPOSITORY'] == tag_name[0] and dic['TAG'] == tag_name[1]  :
+        return True 
+    else :
+        return False
+
+
+
+def main():
+    work_dir = os.getcwd()
+    response = requests.get("https://api.github.com/repos/Samsung/ONE/tags")
+    if  response.status_code != 200 :
+        print("Error was occured while downloading ONE release tags")
+        print("status_code: ", response.status_code)
+        print("Error_reason: " + response.reason)
+        exit(1)
+    
+    versions_str = response.content
+    versions_json = json.loads(versions_str)
+    recent_version = versions_json[0]["name"]
+
+    image_name = f"onecc:{recent_version}"
+    build_arg = f"VERSION={recent_version}"
+    if not _is_image(recent_version):
+        build_cmd = ["docker", "build", "-t", image_name, "--build-arg", build_arg, work_dir]
+        subprocess.call(build_cmd)
+
+    contianer_name = f"onecc_{recent_version.replace('.','_')}"
+    user_cmd = ''.join(sys.argv[1:])
+    run_cmd = ["docker", "run", "--rm", "--name", contianer_name, "-v", "${HOME}:${HOME}", "-e", "HOME=${HOME}", "-w", work_dir, image_name, user_cmd]
```

Comment:
@jyoungyun I'm not sure about mounting host's HOME directory. How about receive volume path from user or use temporary dir if path is not given?

Reply:
> I'm not sure about mounting host's HOME directory


The reason why we mounted `home` is because we considered the following cases.

**User command** 
```
~/onecc-docker/test$ onecc-docker -C ../onecc.template.cfg 
```
**onecc.template.cfg** 
```
...
[one-import-tflite]
# mandatory
; tflite file
input_path=../inception_v3.tflite
; circle file
output_path=inception_v3.circle
...
```
**file structure** 
```
~/onecc-docker$ tree5
.
├── Dockerfile
├── inception_v3.pb
├── inception_v3.tflite
├── onecc-docker
├── onecc.template.cfg
├── onecc_028.workflow.json
└── test
    ├── inception_v3.circle
    ├── inception_v3.circle.log
    ├── inception_v3.opt.circle
    ├── inception_v3.opt.circle.log
    ├── inception_v3_pack
    │   └── inception_v3.opt
    │       ├── inception_v3.opt.circle
    │       └── metadata
    │           └── MANIFEST
    ├── inception_v3_pack.log
    └── onecc-docker

4 directories, 14 files
```

As in the example above, I thought that the user did not know where to receive the input or where to expose the output.

> How about receive volume path from user or use temporary dir if path is not given?

Also, it is hard to understand your ungiven path. Can you explain in detail?

I don't know if that's the right answer, I'm sorry.

Reply:
> Also, it is hard to understand your ungiven path. Can you explain in detail?

I thought below option. But temporary directory things turns out to be invalid. I used `pwd` instead.
```bash
$ onecc-docker -C in.cfg -w my_dir # -w (--working_dir)
# docker run .. -v my_dir:my_dir
$ onecc-docker -C in.cfg # working diectory not given
# docker run .. -v $(pwd):$(pwd)
```

> The reason why we mounted `home` is because we considered the following cases.

Well, I understood what you were trying to say. But I'm not sure it is safe or not. We need a discussion about this.

@jyoungyun Is it okay with mounting users' HOME directory?

Reply:
> Could you add user and group id options so that the result generated in the docker image can be the same user as the host?


@jyoungyun I understand what you are saying, but there are difficulties in the implementation process. Do you mean to modify the docker file to add information about uid and gid?
Here's what I did, but I don't think this is right.

```
docker run --rm --name {container_name} -u 1000 -v ${HOME}:${HOME} -e HOME=${HOME} -w ${PWD} onecc:1.21.0 

$ docker exec {container_name} id
uid=1000 gid=0(root) groups=0(root)
```

Reply:
You can use `id -u` for user id and `id -g` for group id. :)

Reply:
https://medium.com/redbubble/running-a-docker-container-as-a-non-root-user-7d2e00f8ee15

This link will help you :)

Reply:
I checked it well. Thank you!

Reply:
> You can use `id -u` for user id and `id -g` for group id. :)

I am using virtualbox to test our script and using mounted folder with Window and Linux.
Ubuntu user who want to accpet mounted folder need to belong to vboxsf group.
but when using only id -u and id -g option on docker , like docker run -u $(id -u):$(id -g), vboxsf is not inserted into docker user's group so onecc-docker can't access mounted folder

@jyoungyun 
Shoud I insert all groups to docker user's group to care the situation like this or don't care about it and just use docker run -u $(id -u):$(id -g)?

Reply:
> You can use `id -u` for user id and `id -g` for group id. :)

I am using virtualbox to test our script and using mounted folder with Window and Ubuntu.
Ubuntu user who want to accpet mounted folder need to belong to vboxsf group.
but when using only id -u and id -g option on docker , like docker run -u $(id -u):$(id -g), vboxsf is not inserted into docker user's group so onecc-docker can't access mounted folder

@jyoungyun
Shoud I insert all current user's groups to docker user's group to care the situation like this or don't care about it and just use docker run -u $(id -u):$(id -g)?

Reply:
It is cautious to put all user groups. Even if it's a little inconvenient for you to test, what about keeping the existing code?
And in your case, can you change the user id of the mounted folder to ubuntu user?

Reply:
Ok I'll keep existing code! Thank you!